### PR TITLE
Export hackathon tests

### PR DESCRIPTION
**Pull request by richard015ar**
Issue: https://github.com/pressbooks/pressbooks/issues/2448

This PR adds unit tests for the new blade templates end export helpers class.

**Note: We have an validation issue for EPUB export routine, which will be addressed in: https://github.com/pressbooks/pressbooks/issues/2470. So, a few tests related to export fail, not because the implementation but because validation issues.**

### CODE REVIEW

File: inc/modules/export/xhtml/class-xhtml11.php [hunk 1]
```diff
@@ -1517,7 +1517,7 @@ protected function echoBackMatterGenerator( $book_contents, $metadata ) : \Gener
 			] );
 
 			echo $this->blade->render( 'export/generic-post-type', $data );
-			
+
```
Comment (richard015ar) [inline inc/modules/export/xhtml/class-xhtml11.php:1520]: It was a linting auto-fix. For some reason PHPCS complains about this.

File: templates/export/generic-post-type.blade.php [hunk 2]
```diff
@@ -16,9 +16,9 @@
 		{!! $append_post_content !!}
 	@endif
 	@if( isset( $endnotes ) )
-		{{ $endnotes }}
+		{!! $endnotes !!}
```
Comment (richard015ar) [inline templates/export/generic-post-type.blade.php:19]: We can have footnotes and endnotes with paragraph, images, and other html elements.

File: tests/test-modules-export-export.php [hunk 3]
```diff
@@ -448,7 +448,8 @@ public function test_sanityCheckXhtmlWithoutBuckram() {
 		$this->assertContains( '<p><em>Ka kite ano!</em></p>', $xhtml_content );
 		$this->assertContains( 'https://github.com/pressbooks/pressbooks', $xhtml_content );
 		// Heading elements should be in a "bad" place.
-		$this->assertContains( '</h2></div><div class="ugc chapter-ugc"><h2 class="chapter-subtitle">Or, A Chapter to Test</h2>', $xhtml_content );
```
Comment (richard015ar) [inline tests/test-modules-export-export.php:451]: We check structure in blade template testing.

**Review by fdalcin — CHANGES_REQUESTED** [review]
Really like your approach to the new tests @richard015ar.

However, I'm noticing a couple of things: 
I think you've accidentally deleted `tests/data/log.csv`. I left a couple of comments for the tests that fail. 

We have a tests that fails for EPUB validation that because it's rendering the slug in a bad way. We can ignore this failing test for now because this is fixed in another branch and should be fine once both are merged.

**Review by fdalcin — APPROVED** [review]

**Review by fdalcin — APPROVED** [review]
LGMT, thanks @richard015ar

**Pull request merged**